Flexa-Agents/large-withdrawal/src/agent.spec.ts

```diff
@@ -0,0 +1,146 @@
+import { Finding, FindingSeverity, FindingType, HandleTransaction, TransactionEvent } from "forta-agent";
+import { createAddress, TestTransactionEvent } from "forta-agent-tools";
+import { provideHandleTransaction } from "./agent";
+import { when } from "jest-when";
+import { BigNumber } from "ethers";
+import { keccak256 } from "forta-agent/dist/sdk/utils";
+import { AMP_ABI } from "./utils";
+import { Interface } from "@ethersproject/abi";
+
+const AMP_IFACE: Interface = new Interface(AMP_ABI);
+
+const PARTITIONS: string[] = [
+    keccak256("part1"),
+    keccak256("part2"),
+    keccak256("part3"),
+    keccak256("part4"),
+    keccak256("part5"),
+    keccak256("part6"),
+];
+
+const transferByPartitionFinding = (fromPartition: string, fromAddress: string, toAddress: string, value: string): Finding => Finding.fromObject({
+  name: "Large transfer from a Flexa Collateral Manager partition detected",
+  description: "TransferByPartition event emitted with a large value",
+  alertId: "FLEXA-2",
+  severity: FindingSeverity.Info,
+  type: FindingType.Info,
+  protocol: "Flexa",
+  metadata: { fromPartition, fromAddress, toAddress, value },
+});
+
+describe("Forta Collateral Manager large transfer by partition agent test suite", () => {
+    const mockIsPartition = jest.fn();
+    const FLEXA_CONTRACT: string = createAddress("0xfffd");
+    const AMP_CONTRACT: string = createAddress("0xffff");
+    const mockFetcher = {
+        isPartition: mockIsPartition,
+        flexa: FLEXA_CONTRACT,
+        amp: AMP_CONTRACT
+    }
+    const THRESHOLD: BigNumber = BigNumber.from("1000");
+    const handler: HandleTransaction = provideHandleTransaction(mockFetcher as any, THRESHOLD);
+
+    beforeEach(() => {
+        mockFetcher.isPartition.mockClear();
+    });
+
+    it("should report no findings for txs without events", async () => {
+        const txEvent: TransactionEvent = new TestTransactionEvent();
+        const findings: Finding[] = await handler(txEvent);
+        expect(findings).toStrictEqual([]);           
+    });
+
+    it("should ignore regular, below the threshold, transfers", async () => {
+        when(mockIsPartition)
+            .calledWith(38, PARTITIONS[1]).mockReturnValue(true);
+            
+        const event = AMP_IFACE.getEvent("TransferByPartition");
+        const log = AMP_IFACE.encodeEventLog(
+            event, [PARTITIONS[1], createAddress('0x4567'),  mockFetcher.flexa, createAddress("0x7676"), "324", keccak256("k6543"), keccak256("d4949d")]
+        );
+        
+        const txEvent: TransactionEvent =  new TestTransactionEvent()
+            .setBlock(38)
+            .addAnonymousEventLog(mockFetcher.amp, log.data, ...log.topics);        
+
+        const findings: Finding[] = await handler(txEvent);
+        expect(findings).toStrictEqual([]);
+
+    })
+
+    it("should ignore transfers from a non Flexa partition", async () => {
```

Comment:
I guess the name of this test is not accurate, because `fromPartition` is a Flexa partition. I think the name should be something like "should ignore non-withdrawal transfers" because the `from` is not flexa address so it is not a withdrawal

Reply:
Just for my understanding, can you elaborate on why 
> `fromPartition` is a Flexa partition 
?
Couldn't it be a partition belonging to the `from` address?

Reply:
I guess you are right, but in the test you are registering the partition in your mock as a flexa one, so the reason why the event is discarded is because the `from` is not flexa

Reply:
You're right, I get it now - fixed